src/azure-cli/azure/cli/command_modules/sql/custom.py

```diff
@@ -1583,98 +1585,796 @@ def _db_security_policy_update(
             use_secondary_key)
 
 
-def db_audit_policy_update(
+def _check_audit_policy_state(
+        state,
+        value):
+    return state is not None and state.lower() == value.lower()
+
+
+def _is_audit_policy_state_enabled(state):
+    return _check_audit_policy_state(state, BlobAuditingPolicyState.enabled.value)
+
+
+def _is_audit_policy_state_disabled(state):
+    return _check_audit_policy_state(state, BlobAuditingPolicyState.disabled.value)
+
+
+def _is_audit_policy_state_none_or_disabled(state):
+    return state is None or _check_audit_policy_state(state, BlobAuditingPolicyState.disabled.value)
+
+
+def _get_diagnostic_settings_url(
         cmd,
-        instance,
+        resource_group_name,
+        server_name,
+        database_name=None):
+
+    from azure.cli.core.commands.client_factory import get_subscription_id
+
+    return '/subscriptions/{}/resourceGroups/{}/providers/Microsoft.Sql/servers/{}/databases/{}'.format(
+        get_subscription_id(cmd.cli_ctx),
+        resource_group_name, server_name,
+        database_name if database_name is not None else "master")
+
+
+def _get_diagnostic_settings(
+        cmd,
+        resource_group_name,
+        server_name,
+        database_name=None):
+    '''
+    Common code to get server or database diagnostic settings
+    '''
+
+    diagnostic_settings_url = _get_diagnostic_settings_url(cmd, resource_group_name, server_name, database_name)
+    azure_monitor_client = get_mgmt_service_client(cmd.cli_ctx, MonitorManagementClient)
```

Comment:
Try to reuse the following function.

https://github.com/Azure/azure-cli/blob/12c1c4c7d049e85d7fcdc4e46b37e45d2f1dab09/src/azure-cli/azure/cli/command_modules/monitor/_client_factory.py#L8-L11